lib/kubernetes-deploy/kubeclient_builder/google_config.rb

```diff
@@ -0,0 +1,42 @@
+require 'googleauth'
+module KubernetesDeploy
+  module KubeclientBuilder
+    class GoogleConfig < Kubeclient::Config
+      def fetch_user_auth_options(user)
+        if user['auth-provider'] && (user['auth-provider']['name'] == 'gcp')
+          { bearer_token: self.get_token }
+        else
+          super
+        end
+      end
+
+      def self.read(filename)
+        new(YAML.load_file(filename), File.dirname(filename))
+      end
+
+      def get_token
+        scopes = ['https://www.googleapis.com/auth/cloud-platform']
+        authorization = Google::Auth.get_application_default(scopes)
+
+        authorization.apply({})
```

Comment:
Well `apply` adds the access token to a set of headers in a hash. Since we just want to recover the token we don't need to pass in anything.